_posts/2025-04-24-secure-mcp-sse-server.adoc

```diff
@@ -0,0 +1,227 @@
+
+image::mcp_server_choose_tool.png[Choose MCP Server tool,align="center"]
+
+Paste the copied access token into the Tool's `Bearer token` field, and request a new MCP SSE session:
+
+image::mcp_server_bearer_token.png[MCP Server Bearer token,align="center"]
+
+Use the `Call` action to make a tool call and get a response which contains the `alice` user name:
+
+image::mcp_server_tool_response.png[MCP Server tool response,align="center"]
+
+[[mcp-inspector]]
+=== Use MCP Inspector
+
+Launch https://modelcontextprotocol.io/docs/tools/inspector[MCP inspector]:
+
+[source,shell]
+----
+npx @modelcontextprotocol/inspector
+----
+
+Next, get and copy the access token as explained in the <<mcp-server-devui>> section above. You can use the same access token that you may have already used when working through that section.
+
+Use this access token to have MCP inspector connect to the Quarkus MCP SSE server:
+
+image::mcp_inspector_connect.png[MCP Inspector Connect,align="center"]
+
+Next, make a `user-name-provider` tool call:
+
+image::mcp_inspector_tool_call.png[MCP Inspector Tool Call,align="center"]
+
+=== Use curl
+
+Finally, let's use `curl` and also learn a little bit how the MCP protocol and MCP SSE transport work. 
+
+First, access the main SSE endpoint without the access token:
+
+[source,shell]
+----
+curl -v localhost:8080/mcp/sse
+----
+
+You will get HTTP 401 error.
+
+Next, get and copy the access token as explained in the <<mcp-server-devui>> section above. You can use the same access token that you may have already used when working through that section.
+
+[source,shell]
+----
+curl -v -H "Authorization: Bearer ey..." localhost:8080/mcp/sse
+----
+
+and get an SSE response such as:
+
+[source,shell]
+----
+< content-type: text/event-stream
+< 
+event: endpoint
+data: /messages/ZTZjZDE5MzItZDE1ZC00NzBjLTk0ZmYtYThiYTgwNzI1MGJ
+----
+
+The current `curl` SSE session must stay open, for it to receive updates related to the dynamically allocated `data` endpoint.
+
+Open another window and use the same access token to post a tool call request to the `data` endpoint:
+
+[source,shell]
+----
+curl -v -H "Authorization: Bearer ey..." -H "Content-Type: application/json" --data @call.json http://localhost:8080/mcp/messages/ZTZjZDE5MzItZDE1ZC00NzBjLTk0ZmYtYThiYTgwNzI1MGJ
+----
+
+where the `call.json` file looks like this:
+
+[source,json]
+----
+{
+  "jsonrpc": "2.0",
+  "id": 2,
+  "method": "tools/call",
+  "params": {
+    "name": "user-name-provider",
+    "arguments": {
+    }
+  }
+}
+----
+
+Now look in the first curl window and observe a response which contains the `alice` user name:
+
+[source,shell]
+----
+event: message
+data: {"jsonrpc":"2.0","id":2,"result":{"isError":false,"content":[{"text":"alice","type":"text"}]}}
+----
+
+== Conclusion
+
+In this blog post, we have explained how you can easily create a secure Quarkus MCP SSE server, obtain a bearer access token and test it in 
```

Comment:
```suggestion
In this blog post, we have explained how you can easily create an authenticated Quarkus MCP SSE server, obtain a bearer access token and test it in 
```

Reply:
in what? Something missing :-)

Reply:
@cescoffier Hi, lol, I probably jumped to some other section, sure, will fix